proxy/src/main/java/com/wavefront/agent/logsharvesting/InteractiveLogsTester.java

```diff
@@ -0,0 +1,84 @@
+package com.wavefront.agent.logsharvesting;
+
+import com.wavefront.agent.PointHandler;
+import com.wavefront.agent.config.ConfigurationException;
+import com.wavefront.agent.config.LogsIngestionConfig;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.Scanner;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
+
+import javax.annotation.Nullable;
+
+import sunnylabs.report.ReportPoint;
+
+/**
+ * @author Mori Bellamy (mori@wavefront.com)
+ */
+public class InteractiveLogsTester {
+
+  private final Supplier<LogsIngestionConfig> logsIngestionConfigSupplier;
+  private final String prefix;
+  private final Scanner stdin;
+
+  public InteractiveLogsTester(Supplier<LogsIngestionConfig> logsIngestionConfigSupplier, String prefix)
+      throws ConfigurationException {
+    this.logsIngestionConfigSupplier = logsIngestionConfigSupplier;
+    this.prefix = prefix;
+    stdin = new Scanner(System.in);
+  }
+
+  /**
+   * Read one line of stdin and print a message to stdout.
+   */
+  public boolean interactiveTest() throws ConfigurationException {
+    final AtomicBoolean reported = new AtomicBoolean(false);
+
+    LogsIngester logsIngester = new LogsIngester(
+        new PointHandler() {
+          @Override
+          public void reportPoint(ReportPoint point, @Nullable String debugLine) {
+            reported.set(true);
+            System.out.println(point.toString());
```

Comment:
Maybe use PointHandlerImpl.pointToString() instead to output data in a format more familiar to the user? That method is begging to be moved elsewhere (or at least be made public).